app/deliver_grant_funding/routes.py

```diff
@@ -1,35 +1,127 @@
+
+    return render_template("deliver_grant_funding/grant_setup/ggis_number.html", form=form)
+
+
+@deliver_grant_funding_blueprint.route("/grants/setup/name", methods=["GET", "POST"])
+@platform_admin_role_required
+def grant_setup_name() -> ResponseReturnValue:
+    if "grant_setup" not in session:
+        return redirect(url_for("deliver_grant_funding.grant_setup_intro"))
+
+    grant_session = GrantSetupSession.from_session(session["grant_setup"])
+    form = GrantNameForm(obj=grant_session)
+
+    if form.validate_on_submit():
+        assert form.name.data is not None, "Grant name must be provided"
+        grant_session.name = form.name.data
+        session["grant_setup"] = grant_session.to_session_dict()
+        return redirect(url_for("deliver_grant_funding.grant_setup_description"))
+
+    return render_template("deliver_grant_funding/grant_setup/name.html", form=form)
+
+
+@deliver_grant_funding_blueprint.route("/grants/setup/description", methods=["GET", "POST"])
+@platform_admin_role_required
+def grant_setup_description() -> ResponseReturnValue:
+    if "grant_setup" not in session:
+        return redirect(url_for("deliver_grant_funding.grant_setup_intro"))
+
+    grant_session = GrantSetupSession.from_session(session["grant_setup"])
+    form = GrantDescriptionForm(obj=grant_session)
+
+    if form.validate_on_submit():
+        assert form.description.data is not None, "Grant description must be provided"
+        grant_session.description = form.description.data
+        session["grant_setup"] = grant_session.to_session_dict()
+        return redirect(url_for("deliver_grant_funding.grant_setup_contact"))
+
+    return render_template("deliver_grant_funding/grant_setup/description.html", form=form)
+
+
+@deliver_grant_funding_blueprint.route("/grants/setup/contact", methods=["GET", "POST"])
+@platform_admin_role_required
 @auto_commit_after_request
-def create_grant() -> ResponseReturnValue:
-    form = GrantForm()
+def grant_setup_contact() -> ResponseReturnValue:
+    if "grant_setup" not in session:
+        return redirect(url_for("deliver_grant_funding.grant_setup_intro"))
+
+    grant_session = GrantSetupSession.from_session(session["grant_setup"])
+    form = GrantContactForm(obj=grant_session)
+
     if form.validate_on_submit():
         try:
-            assert form.name.data is not None
-            interfaces.grants.create_grant(name=form.name.data)
-            return redirect(url_for("deliver_grant_funding.list_grants"))
+            assert form.primary_contact_name.data, "Primary contact name must be provided"
+            assert form.primary_contact_email.data, "Primary contact email must be provided"
+
+            grant_session.primary_contact_name = form.primary_contact_name.data
+            grant_session.primary_contact_email = form.primary_contact_email.data
+
+            grant = interfaces.grants.create_grant(
+                name=grant_session.name or "",
```

Comment:
optional: Are there tweaks to the pydantic model we can do that make this `or` unnecessary? Eg giving the fields a default value of `""` or something?